corehq/motech/repeaters/models.py

```diff
@@ -365,10 +365,17 @@ def register(self, payload):
         )
         metrics_counter('commcare.repeaters.new_record', tags={
             'domain': self.domain,
-            'doc_type': self.doc_type
+            'doc_type': self.doc_type,
+            'mode': 'sync' if fire_synchronously else 'async'
         })
         repeat_record.save()
-        repeat_record.attempt_forward_now()
+
+        if fire_synchronously:
+            # Prime the cache to prevent unnecessary lookup. Only do this for synchronous repeaters
+            # to prevent serializing the repeater in the celery task payload
+            RepeatRecord.repeater.fget.get_cache(repeat_record)[()] = self
```

Comment:
I'm not familiar with `fget`. Is that a couchdbkit property or something? Couldn't find it in the code.

Reply:
This is a bit of a hack that's used in a few places in our code. `fget` is the original function that is passed to the property constructor. From the python docs:

> The returned property object also has the attributes fget, fset, and fdel corresponding to the constructor arguments. 

https://docs.python.org/3/library/functions.html?highlight=fget#property

Reply:
Oh I see, thanks